buidl/test/test_musig.py

```diff
@@ -220,50 +475,88 @@ def test_multi_leaf_multisig_locktime(self):
         ]
         points = [priv.point for priv in private_keys]
         tr_multisig = TapRootMultiSig(points, 3)
-        current_locktime = Locktime(66000)
+        current_locktime = Locktime(74930)
         tap_root_input = tr_multisig.multi_leaf_tap_root(locktime=current_locktime)
-        tap_root_output = tr_multisig.multi_leaf_tap_root(
-            sequence=Sequence.from_relative_blocks(50)
-        )
-        leaf = MultiSigTapScript(points[:3], 3, locktime=current_locktime).tap_leaf()
-        self.assertTrue(tap_root_input.control_block(leaf))
         self.assertEqual(
             tap_root_input.address(network="signet"),
-            "tb1pvl3r0t48sxypr6v938rlkykrzpp36zkee2krqh94lk7lr9utqf8sas0r0g",
+            "tb1p7glmm24y2gqsf58emquv6mzy9m6msnwtenj5s77df33t3cal3wkssw5320",
         )
         prev_tx = bytes.fromhex(
-            "cee7412e4aba98dd9e33a88221fc4752a644aeb3f5f444b3af0bd359b7a4edd9"
+            "03be9cbe084ef9deb54befac24381459ee89fbc9fc46079b9bde71451be95219"
         )
-        tx_in = TxIn(prev_tx, 0, sequence=0xFFFFFFFE)
-        tx_in._value = 9999600
-        tx_in._script_pubkey = tap_root_input.script_pubkey()
-        tx_out = TxOut(tx_in.value() - 204, tap_root_output.script_pubkey())
-        tx_obj = Tx(
-            1, [tx_in], [tx_out], current_locktime, network="signet", segwit=True
+        tx_ins = []
+        for prev_index in range(11):
+            tx_in = TxIn(prev_tx, prev_index, sequence=0xFFFFFFFE)
+            tx_in._value = 1000000
+            tx_in._script_pubkey = tap_root_input.script_pubkey()
+            tx_ins.append(tx_in)
+        change_script = address_to_script_pubkey(
+            "tb1q7kn55vf3mmd40gyj46r245lw87dc6us5n50lrg"
         )
-        tx_in.witness.items = []
-        tx_obj.initialize_p2tr_multisig(
-            0,
-            tap_root_input.control_block(leaf),
-            leaf.tap_script,
+        fee = 1634
+        tx_out = TxOut(1000000 * len(tx_ins) - fee, change_script)
+        tx_obj = Tx(
+            1, tx_ins, [tx_out], current_locktime, network="signet", segwit=True
         )
-        sigs = [
-            tx_obj.get_sig_taproot(0, priv, ext_flag=1) for priv in private_keys[:3]
-        ]
-        tx_obj.finalize_p2tr_multisig(0, sigs)
+        for input_index, pubkeys in enumerate(combinations(points, 3)):
+            leaf = MultiSigTapScript(pubkeys, 3, locktime=current_locktime).tap_leaf()
+            self.assertTrue(tap_root_input.control_block(leaf))
+            tx_in = tx_ins[input_index]
+            tx_in.witness.items = []
+            tx_obj.initialize_p2tr_multisig(
+                input_index,
+                tap_root_input.control_block(leaf),
+                leaf.tap_script,
+            )
+            sigs = []
+            for priv in private_keys:
```

Comment:
Should there be a helper method for this? These ~8 lines for gathering the sigs are regularly used and quite complex for the downstream caller. Could DRY it out.

Reply:
That's basically what `finalize_p2tr_multisig` does. The loop is there for the coordinator to gather the `sigs` array. In practice, the stuff in the for loop will most likely be done by different keys and generated so this is really a way to simulate the signing.

I could DRY out this since it gets repeated, but I've found doing that in tests tends to not be a good idea as changing the helper test function tends to break lots of tests at once.